### PR TITLE
Better NonHashValueOverride error message

### PR DESCRIPTION
**Pull request by WGautier**

### CODE REVIEW

File: lib/environment_config_transmogrifier.rb [hunk 1]
```diff
@@ -75,7 +75,7 @@ def self.inject_value(hash, key_grams, value)
     # error out if we're trying to override
     # an existing value that's not a hash
     unless hash[key_grams[0]].is_a?(Hash)
-      raise NonHashValueOverride, 'Refusing to override non-hash value.'
+      raise NonHashValueOverride, "Refusing to override non-hash value: #{key_grams}"
```
Comment (Mook-as) [inline lib/environment_config_transmogrifier.rb]: This being a recursive function, it'll only log the conflicting part; it definitely helps, but will still be hard to locate. Do we want to make it hold on to the original too?

Comment (Mook-as) [inline lib/environment_config_transmogrifier.rb]: Also, `key_grams` is an array, we probably want to `.join('.')` first.

**Comment by WGautier** [comment]
How about this?

**Comment by Mook-as** [comment]
Code looks good, trying to see it in action now.
(Hint: That means having a test would let me rubber-stamp it :D )

**Comment by WGautier** [comment]
I'm still figuring out how this error occurs, so I will add a test when I do figure it out tomorrow.

**Comment by Mook-as** [comment]
I think I've managed to hit it:

`config_spec.json`:
```json
{"properties": {"foo": 1 } }
```
`env2conf.yml`:
```yaml
properties.foo.bar: value
```
Command line:
```sh
bundle exec bin/configgin -i /dev/null -o /dev/null -b config_spec.json -e env2conf.yml 
```

Message (manually line-wrapped):
```
environment_config_transmogrifier.rb:79:in `inject_value':
   Refusing to override non-hash value: 'foo.bar' -
   Complete key: 'properties.foo.bar' (NonHashValueOverride)
```

I guess the test can just assert that the full key name (`properties.foo.bar` in this case) is in there?

**Pull request merged**